frontend/src/metabase-lib/lib/queries/StructuredQuery.js

```diff
@@ -1124,11 +1124,19 @@ export default class StructuredQuery extends AtomicQuery {
       const keyForFk = (src, dst) =>
         src && dst ? `${src.id},${dst.id}` : null;
       const explicitJoins = new Set(
```

Comment:
`dimensionOptions` is awfully complex. can you pull out the logic for getting the `explicitJoins` set?

the `getExplicitJoinsSet` function (or whatever you want to call it) doesn't necessarily need to live on `StructuredQuery` either, so maybe think about sticking the function elsewhere, too.

Reply:
Good point. I don't have a solid idea yey on how and where should we keep `metabase-lib` "utilities", so I'll try to just extract that in a new method for now